python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+    -------
+    wcs : `wcsfit.Wcs`
+        WCS object in TPV format
+    """
+    fits_metadata = butlerWcs.getFitsMetadata()
+    if not ((fits_metadata.get('CTYPE1') == 'RA---TAN-SIP')
+            and (fits_metadata.get('CTYPE2') == 'DEC--TAN-SIP')):
+        raise ValueError(f'CTYPES {fits_metadata.get("CTYPE1")} and {fits_metadata.get("CTYPE2")}'
+                         'do not match SIP convention')
+
+    floatDict = {k: fits_metadata[k] for k in fits_metadata if isinstance(fits_metadata[k], (int, float))}
+
+    wcs = wcsfit.readTPVFromSIP(floatDict, "SIP")
+
+    return wcs
+
+
+class WCSFitConnections(pipeBase.PipelineTaskConnections,
+                        dimensions=("skymap", "tract", "instrument", "physical_filter")):
+    """Middleware input/output connections for task data."""
+    inputCatalogRefs = pipeBase.connectionTypes.Input(
+        doc="Source table in parquet format, per visit",
+        name="preSourceTable_visit",
+        storageClass="DataFrame",
+        dimensions=("instrument", "visit"),
+        deferLoad=True,
+        multiple=True,
+    )
+    inputVisitSummary = pipeBase.connectionTypes.Input(
+        doc=("Per-visit consolidated exposure metadata built from calexps. "
+             "These catalogs use detector id for the id and must be sorted for "
+             "fast lookups of a detector."),
+        name="visitSummary",
+        storageClass="ExposureCatalog",
+        dimensions=("instrument", "visit"),
+        multiple=True,
+    )
+    referenceCatalog = pipeBase.connectionTypes.PrerequisiteInput(
+        doc="The astrometry reference catalog to match to loaded input catalog sources.",
+        name="gaia_dr2_20200414",
+        storageClass="SimpleCatalog",
+        dimensions=("skypix",),
+        deferLoad=True,
+        multiple=True,
+        lookupFunction=lookupVisitRefCats,
+    )
+    outputWcs = pipeBase.connectionTypes.Output(
+        doc=("Per-tract, per-visit world coordinate systems derived from the fitted model."
+             " These catalogs only contain entries for detectors with an output, and use"
+             " the detector id for the catalog id, sorted on id for fast lookups of a detector."),
+        name="WCSFitSkyWcsCatalog",
+        storageClass="ExposureCatalog",
+        dimensions=("instrument", "visit", "skymap", "tract"),
+        multiple=True
+    )
+    outputCatalog = pipeBase.connectionTypes.Output(
+        doc=("Source table with stars used in fit, along with residuals in pixel coordinates and tangent "
+             "plane coordinates and chisq values"),
+        name="WCSFit_usedStars",
+        storageClass="DataFrame",
+        dimensions=("instrument", "skymap", "tract"),
+    )
+
+
+class WCSFitConfig(pipeBase.PipelineTaskConfig,
+                   pipelineConnections=WCSFitConnections):
+    """Configuration for WCSFitTask"""
+    sourceSelector = sourceSelectorRegistry.makeField(
+        doc="How to select sources for cross-matching",
```

Comment:
doc strings in configs should have a trailing period.  Also above for connections.

Reply:
This makes me unhappy because the doc strings are not sentences. Do you know where it is in the dev guide so I can try to understand this rule?

Reply:
Field docstrings should be sentences (potentially multiple); they are built into the `Configuration Fields` section of the pipeline docs. Our examples in `pex.Config` are inconsistent on punctuation and capitalization: I'll file a ticket to get that cleaned up.

Reply:
https://jira.lsstcorp.org/browse/DM-36696